### PR TITLE
Add gitignore

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR adds .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+*.so
+*.dylib
+
+# Test binary, build with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+### Terraform ###
```
Comment (enxebre) [inline .gitignore]: curious why considering any terraform related?

Comment (alexander-demichev) [reply .gitignore]: No reason actually, I copied this file from AWS https://github.com/openshift/cluster-api-provider-aws/blob/master/.gitignore

Comment (enxebre) [reply .gitignore]: We use to run terraform for some testing there. I don't think we need this now.

**Pull request merged**